### PR TITLE
minecraftia: init at 1.0

### PR DESCRIPTION
**Pull request by gepbird**
## Description of changes
Add Minecraftia.ttf, the font used in the game Minecraft.
Description for this package was taken from the host website: https://fontlibrary.org/en/font/minecraftia

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by h7x4 — APPROVED** [review]
First time in quite a while I haven't found anything to comment on. Good work!

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package build on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden
- [X] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [X] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [X] patches that are remotely available are fetched rather than vendored


Result of `nixpkgs-review pr 261705` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>minecraftia</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -29651,6 +29651,8 @@ with pkgs;
 
   migu = callPackage ../data/fonts/migu { };
 
+  minecraftia = callPackage ../data/fonts/minecraftia { };
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:29654]: Please move the package to `pkgs/by-name/mi/minecraftia/package.nix` and drop this line, thanks.

File: pkgs/data/fonts/minecraftia/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, fetchzip, stdenvNoCC }:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (wegank) [inline pkgs/data/fonts/minecraftia/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/data/fonts/minecraftia/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "minecraftia";
+  version = "1.0";
```
Comment (wegank) [inline pkgs/data/fonts/minecraftia/default.nix:5]: Where does this version come from?

Comment (gepbird) [reply pkgs/data/fonts/minecraftia/default.nix:5]: I checked the font's metadata using Birdfont. It is very possible that the author didn'tt care about versioning and went with 1.0 as the default. As far as I'm aware there are no other versions of this font.

**Review by wegank — APPROVED** [review]

**Pull request merged**